gaia-dialog.js

```diff
@@ -28,127 +41,108 @@ module.exports = component.register('gaia-dialog', {
-    debug('animate in from target');
-    var pos = e.touches && e.touches[0] || e;
-    var scale = Math.sqrt(window.innerWidth * window.innerHeight) / 10;
-    var background = this.els.background;
-    var duration = scale * 7;
-    var end = 'transitionend';
-    var self = this;
-
-    background.style.transform =
-      'translate(' + pos.clientX + 'px, ' + pos.clientY + 'px)';
-    background.style.transitionDuration = duration + 'ms';
-    background.classList.add('circular');
-    debug('animation start');
-    this.dispatch('animationstart');
-
-    background.offsetTop;
-
-    background.style.transform += ' scale(' + scale + ')';
-    background.style.opacity = 1;
-
-    background.addEventListener(end, function fn() {
-      background.removeEventListener(end, fn);
-      debug(end);
-      self.els.window.classList.add('animate-in');
-      self.dispatch('animationend');
-    });
+  animateBackgroundIn(options) {
+    if (options) { return this.animateBackgroundInFrom(options); }
+
+    var el = this.els.background;
+    return schedule.transition(() => {
+      debug('animate background in');
+      el.classList.remove('animate-out');
+      el.classList.add('animate-in');
+    }, el, 'animationend');
   },
 
-  animateOut: function(callback) {
-    var end = 'animationend';
-    var background = this.els.background;
-    var self = this;
-    var classes = {
-      el: this.classList,
-      window: this.els.window.classList,
-      background: this.els.background.classList
-    };
+  animateBackgroundOut() {
+    var el = this.els.background;
+    return schedule.transition(() => {
+      debug('animate background out');
+      el.classList.add('animate-out');
+      el.classList.remove('animate-in');
+    }, el, 'animationend')
+      .then(() => el.style = '');
+  },
 
-    this.dispatch('animationstart');
-    classes.window.add('animate-out');
-
-    this.els.window.addEventListener(end, function fn(e) {
-      self.els.window.removeEventListener(end, fn);
-      e.stopPropagation();
-      classes.background.add('animate-out');
-      self.els.background.addEventListener(end, function fn() {
-        self.els.background.removeEventListener(end, fn);
-        classes.window.remove('animate-out', 'animate-in');
-        classes.background.remove('animate-out', 'animate-in');
-        background.classList.remove('circular');
-        background.style = '';
-        self.dispatch('animationend');
-        if (callback) { callback(); }
+  animateBackgroundInFrom(pos) {
+    var el = this.els.background;
+    var scale = Math.sqrt(window.innerWidth * window.innerHeight) / 15;
+    var duration = scale * 9;
+
+    return schedule.mutation(() => {
+        el.classList.add('circular');
+        el.classList.remove('animate-out');
+        el.style.transform = `translate(${pos.clientX}px, ${pos.clientY}px)`;
+        el.style.transitionDuration = duration + 'ms';
+        el.offsetTop; // Hack, any ideas?
```

Comment:
Without it we skip the transition and  jumps to final frame. Need to make sure we get one paint in start styles before applying end styles.